site/docs/v1/tech/tutorials/integrate-laravel-api.md

```diff
@@ -0,0 +1,245 @@
+{% remote_include https://raw.githubusercontent.com/FusionAuth/fusionauth-example-laravel-api/main/laravel/app/FusionAuth/FusionAuthJWTGuard.php %}
+```
+
+To override the existing classes when the ones you just added, you must create a [Service Provider](https://laravel.com/docs/10.x/providers) in `app/FusionAuth/Providers/FusionAuthServiceProvider.php`.
+
+```php
+{% remote_include https://raw.githubusercontent.com/FusionAuth/fusionauth-example-laravel-api/main/laravel/app/FusionAuth/Providers/FusionAuthServiceProvider.php %}
+```
+
+Finally, expose your Service Provider by adding it to the `providers` array in `config/app.php`.
+
+```diff
+{% remote_include https://github.com/FusionAuth/fusionauth-example-laravel-api/commit/b291247bc0373cffa4edc1a64a58e475f706ff40.diff %}
+```
+
+### Retrieving Public Key
+
+To make your API trust JWTs issued by FusionAuth, you must import the Public Key from your FusionAuth application into it.
+
+If you have [jq](https://stedolan.github.io/jq/download/) _(a script to parse JSON objects)_ installed, you can run the command below to fetch and save it to `storage/public-key.pem`.
+
+```shell
+curl -H 'Authorization: this_really_should_be_a_long_random_alphanumeric_value_but_this_still_works' http://localhost:9011/api/key/1afa4d7e-76f0-45e9-bb46-98be5329ef37 | jq -r '.key.publicKey' > storage/public-key.pem
+```
+
+If you don't have it, log into the [FusionAuth admin screen](http://localhost:9011) using the admin user credentials ("admin@example.com"/"password"), navigate to <span>Settings -> Key Master</span>{:.breadcrumb}, locate the key named `For exampleapp` and click its download button. Inside the downloaded `.zip` file, go to the `keys` folder and extract `public-key.pem` to the `storage` folder in your Laravel app.
+
+### Creating a Controller
+
+Create a controller in `app/Http/Controllers/MessagesController.php` which gives back a JSON message.
+
+```php
+{% remote_include https://raw.githubusercontent.com/FusionAuth/fusionauth-example-laravel-api/main/laravel/app/Http/Controllers/MessagesController.php %}
+```
+
+Add a route to `routes/api.php` so it can be possible to make a `GET` request to `/api/messages`.
+
+```diff
+{% remote_include https://github.com/FusionAuth/fusionauth-example-laravel-api/commit/3fd47ffc4942190c7289e961133baf74c56b5e53.diff %}
+```
+
+### Testing the Authentication Flow
+
+Finally, start your application.
+
+```shell
+./vendor/bin/sail up -d
+```
+
+Make sure that everything is running fine by making a request to [localhost/api/auth/me](http://localhost/api/auth/me).
+
+```shell
+curl -X POST -H 'Accept: application/json' http://localhost/api/auth/me
+```
+
+You should see an error message because you're not providing any JWT to the request.
+
+```json
+{"message":"Unauthenticated."}
+```
+
+Now, you need to call FusionAuth to get an access token. For ease of use, these instructions will use the Login API, but you could also get the access token via the hosted login pages.
```

Comment:
Would love to grab the html code from teh c# api quickstart: https://github.com/FusionAuth/fusionauth-site/pull/2225 starting with Create a Login Redirect Page and use that. I think it shows the power of the hosted backend, as long as the jwt guard can read from a cookie.

Reply:
Done. I think it's looking neat :D 

![Peek 2023-06-14 16-00](https://github.com/FusionAuth/fusionauth-site/assets/1877191/0138a06e-f391-46c4-9936-9abc240ff52a)

Reply:
That is awesome.